proc-macros/src/rpc_macro.rs

```diff
@@ -42,6 +42,7 @@ pub struct RpcMethod {
 	pub blocking: bool,
 	pub docs: TokenStream2,
 	pub params: Vec<(syn::PatIdent, syn::Type)>,
+	pub param_format: String,
```

Comment:
I think I would prefer an `Option`,  to not regard invalid param format as `array`

such as
```
let params = match parms {
  None => Array,
  Some("array") => Array,
  Some("map") => Map
  err => compile_error!("Invalid param {:?} expected: `array` or `map`", err),
};
```